framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -102,29 +114,28 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 
 	public async load(): Promise<void> {
 		this._chainConnectorPluginDB = await getDBInstance(this.dataPath);
-		this._sidechainChainConnectorStore = new ChainConnectorStore(
-			this._chainConnectorPluginDB,
-			DB_KEY_SIDECHAIN,
-		);
-		this.endpoint.load(this._sidechainChainConnectorStore);
+		this._chainConnectorStore = new ChainConnectorStore(this._chainConnectorPluginDB);
+		this.endpoint.load(this._chainConnectorStore);
 
-		this._mainchainAPIClient = await apiClient.createIPCClient(this.config.mainchainIPCPath);
-		if (this.config.sidechainIPCPath) {
-			this._sidechainAPIClient = await apiClient.createIPCClient(this.config.sidechainIPCPath);
+		this._receivingChainClient = await apiClient.createIPCClient(this.config.receivingChainIPCPath);
+		if (this.config.sendingChainIPCPath) {
+			this._sendingChainClient = await apiClient.createIPCClient(this.config.sendingChainIPCPath);
 		} else {
-			this._sidechainAPIClient = this.apiClient;
+			this._sendingChainClient = this.apiClient;
 		}
 
 		this._ownChainID = Buffer.from(
 			(
-				await this._sidechainAPIClient.invoke<OwnChainAccountJSON>(
+				await this._sendingChainClient.invoke<OwnChainAccountJSON>(
 					'interoperability_getOwnChainAccount',
 				)
 			).chainID,
 			'hex',
 		);
+		// If the running node is mainchain then receiving chain will be sidehchain or vice verse.
+		this._isReceivingChainIsMainchain = !getMainchainID(this._ownChainID).equals(this._ownChainID);
 		// Fetch last certificate from the receiving chain and update the _lastCertificate
-		const { lastCertificate } = await this._mainchainAPIClient.invoke<ChainAccountJSON>(
+		const { lastCertificate } = await this._receivingChainClient.invoke<ChainAccountJSON>(
```

Comment:
Is there specific reason why only lastCertificate is kept in memory rather than adding it to the DB?

Reply:
lastCertificate is always retrieved when we load and there is going to be only one record. I think here we can remove it from load function and keep it undefined as while calculate CCU params we always call this again and get the latest lastCertificate